src/routes/InventoryOrEdgeComponent.js

```diff
@@ -0,0 +1,52 @@
+import React, { useEffect, useState } from 'react';
+import EdgeGroupsView from '../components/InventoryGroups/EdgeGroups';
+import InventoryGroups from '../components/InventoryGroups/InventoryGroups';
+import { inventoryHasConventionalSystems } from '../Utilities/conventional';
+import { inventoryHasEdgeSystems } from '../Utilities/edge';
+import useEdgeGroups from '../Utilities/hooks/useEdgeGroups';
+import useFeatureFlag from '../Utilities/useFeatureFlag';
+import PropTypes from 'prop-types';
+import { Spinner } from '@patternfly/react-core';
+
+const InventoryOrEdgeView = () => {
+  const [enforceEdgeGroups, isLoading] = useEdgeGroups();
+  const [hasConventionalSystems, setHasConventionalSystems] = useState(true);
+  const [hasEdgeDevices, setHasEdgeDevices] = useState(true);
+  const edgeParityInventoryListEnabled = useFeatureFlag(
+    'edgeParity.inventory-list'
+  );
+
+  useEffect(() => {
+    try {
+      (async () => {
+        const hasConventionalSystems = await inventoryHasConventionalSystems();
+        if (edgeParityInventoryListEnabled) {
+          const hasEdgeSystems = await inventoryHasEdgeSystems();
+          setHasConventionalSystems(hasConventionalSystems);
+          setHasEdgeDevices(hasEdgeSystems);
+        }
+      })();
+    } catch (e) {
+      console.log(e);
+    }
+  }, []);
+
+  const ViewComponent = enforceEdgeGroups ? EdgeGroupsView : InventoryGroups;
+  if (!isLoading) {
+    return (
+      <ViewComponent
+        hasConventionalSystems={hasConventionalSystems}
+        hasEdgeDevices={hasEdgeDevices}
```

Comment:
Do we really need these props here? Neither EdgeGroupsView nor InventoryGroups component accepts these two props. I tried to trace EdgeGroupsView federated module interface and found this https://github.com/RedHatInsights/edge-frontend/blob/master/src/Routes/Groups/Groups.js#L21: it only accepts `locationProp, navigateProp` props.

Reply:
it doesnt work without that